src/main/java/emu/grasscutter/game/managers/EnergyManager/EnergyManager.java

```diff
@@ -327,6 +329,10 @@ public void handleAttackHit(EvtBeingHitInfo hitInfo) {
      Energy logic related to using skills.
      **********/
     private void handleBurstCast(Avatar avatar, int skillId) {
+        if (avatar.getSkillDepot() != null && skillId == avatar.getSkillDepot().getEnergySkill()) {
+            player.setPlayerElementBurstInvincibleEndTime(TimeUnit.NANOSECONDS.toMillis(System.nanoTime()) - player.getSession().getLag() + TimeUnit.MILLISECONDS.toMillis(2000));
+        }
+
```

Comment:
This really does not belong in `EnergyManager`, given that it has nothing to do with energy.

Reply:
Yes, you are right. I will optimize for that.